pages/api/auth/agent-connect/callback.ts

```diff
@@ -10,6 +14,26 @@ export default withSession(async function callbackRoute(req, res) {
   try {
     const userInfo = await agentConnectAuthenticate(req);
     const agent = await getAgent(userInfo);
+
+    const isWhitelisted = agent.scopes.some((scope) => isAgentScope(scope));
+    const { isMCP } = agent;
+
+    if (!isWhitelisted && isMCP) {
+      const siren = extractSirenFromSiret(agent.siret);
+      const uniteLegale = await clientUniteLegaleRechercheEntreprise(siren, 0);
+
+      const isNotServicePublic = !isServicePublic(uniteLegale);
+      // TODO filter base on uniteLegal if it's not a service public for sure
+      const couldBeServicePublic = true;
```

Comment:
As a first implementation, I suggest to check on code juridique. As far as I am concern, ambiguous codes are 4xx and 8xx :  
```
'4110':"Établissement public national à caractère industriel ou commercial doté d'un comptable public",
'4120':"Établissement public national à caractère industriel ou commercial non doté d'un comptable public",
'4130': 'Exploitant public',
'4140': 'Établissement public local à caractère industriel ou commercial',
'4150':
"Régie d'une collectivité locale à caractère industriel ou commercial",
'4160': 'Institution Banque de France',
&
'8110': 'Régime général de la Sécurité Sociale',
'8120': 'Régime spécial de Sécurité Sociale',
'8130': 'Institution de retraite complémentaire',
'8140': 'Mutualité sociale agricole',
'8150': 'Régime maladie des non-salariés non agricoles',
'8160': 'Régime vieillesse ne dépendant pas du régime général de la Sécurité Sociale',
'8170': "Régime d'assurance chômage",
'8190': 'Autre régime de prévoyance sociale',
'8210': 'Mutuelle',
'8250': 'Assurance mutuelle agricole',
'8290': 'Autre organisme mutualiste',
'8310': 'Comité social économique d’entreprise',
'8311': "Comité social économique d'établissement",
'8410': 'Syndicat de salariés',
'8420': 'Syndicat patronal',
'8450': 'Ordre professionnel ou assimilé',
'8470': 'Centre technique industriel ou comité professionnel du développement économique',
'8490': 'Autre organisme professionnel',
'8510': 'Institution de prévoyance',
'8520': 'Institution de retraite supplémentaire',
```

This should be checked in a helper that can be unit tested